### PR TITLE
PR: Fix bug in preferences with max line length

### LINKED ISSUE #14175: Allow setting line length, when setting `black` as autoformatter
## Problem Description

When setting `black` as autoformatter, the line-length (and the vertical ruler) are set and fixed to `88`.
Related: #14082

### What steps reproduce the problem?

1. set `Preferences` -> `Completion and linting` -> `Code formatting` -> `Choose the code formatting provider` to `black`
2. `Preferences` -> `Completion and linting` -> `Code style` -> `Maximum allowed line length` is auto. set to `88`
3. can't change this `Maximum allowed line length`

### What is the expected output? What do you see instead?
Preset of `88` for `black` is fine, but since `black` provides the `-l` option ([link](https://github.com/psf/black#command-line-options)), the maximum line length should be changeable. 

## Versions
* Spyder version: 4.2.0

### Dependencies
* black

### PR DESCRIPTION
**Pull request by steff456**
## Description of Changes

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

Remove the created constraints of the max line length when changing the autoformatter engine and move the `maximum allowed line length` outside the code style linting group.

image

### Issue(s) Resolved

Fixes #14175 

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct: Steff456

### CODE REVIEW

**Comment by steff456** [comment]
image

File: spyder/plugins/completion/languageserver/confpage.py [hunk 1]
```diff
@@ -240,16 +236,20 @@ def setup_page(self):
...
         code_style_layout.addWidget(code_style_label)
         code_style_layout.addWidget(self.code_style_check)
         code_style_layout.addWidget(code_style_g_widget)
-        code_style_layout.addWidget(vertical_line_box)
+        code_style_group.setLayout(code_style_layout)
 
-        code_style_widget = QWidget()
-        code_style_widget.setLayout(code_style_layout)
+        # Maximum allowed line length layout
+        line_length_group = QGroupBox(_("Maximum allowed line length"))
```
Comment (ccordoba12) [inline spyder/plugins/completion/languageserver/confpage.py:247]:
```suggestion
        line_length_group = QGroupBox(_("Line length"))
```

Just to not be repeat the text below.

**Review by ccordoba12 — COMMENTED** [review]
As part of this PR, please also change the text of the checkbox:

`Show vertical line at maximum allowed length`

to

`Show vertical line at that length`

**Review by ccordoba12 — APPROVED** [review]
Thanks @steff456! This is a really great improvement!

**Pull request merged**